driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -362,14 +362,22 @@ public MongoCredential withMechanism(final AuthenticationMechanism mechanism) {
      * @param source    the source of the user name, typically a database name
      * @param password  the password
      */
-    MongoCredential(@Nullable final AuthenticationMechanism mechanism, @Nullable final String userName, final String source,
-                    @Nullable final char[] password) {
+    MongoCredential(
+            @Nullable final AuthenticationMechanism mechanism,
```

Comment:
I don't think this is our coding standard now (one parameter per line).

Reply:
Could we consider making this a coding standard? That is, if there are enough parameters that the line is broken, it should be broken per parameter.

The current form obscures the number of parameters, their details, their correspondence to fields, and how signatures compare to their overloads. It seems equivalent in legibility to putting field declarations on the same line.

Reply:
Sure, we can consider it.  But it's not in scope.